src/act.offensive.cpp

```diff
@@ -598,7 +598,19 @@ bool passed_flee_success_check(struct char_data *ch) {
   if (!can_hurt(ch, FIGHTING(ch), TRUE, 0))
     return TRUE;
 
-  return success_test(GET_QUI(ch), GET_QUI(FIGHTING(ch))) > 0;
+  if success_test(GET_QUI(ch), GET_QUI(FIGHTING(ch)) + racial_flee_modifier) > 0
+    switch (GET_RACE(ch)) {
+      case RACE_SATYR:
+       racial_flee_modifier--;
+       break;
+      case RACE_DWARF:
+      case RACE_GNOME:
+      case RACE_MENEHUNE:
+        racial_flee_modifier++;
+        break;
+        }
+    return TRUE;
```

Comment:
This statement has fallen outside of the if-check due to not using a block statement, so the effect of this code is to always return true and guarantee flee success!